python/pyarrow/tests/test_dataset.py

```diff
@@ -3074,6 +3074,28 @@ def test_orc_format_not_supported():
             ds.dataset(".", format="orc")
 
 
+def test_orc_writer_not_implemented_for_dataset():
+    try:
+        from pyarrow.dataset import OrcFileFormat  # noqa
+    except ImportError:
+        pytest.skip('this case is covered by test_orc_format_not_supported')
```

Comment:
You could also skip the test with a pytest mark, like here: https://github.com/apache/arrow/blob/f67009aac21e8e615743bbc2933fdbf8376953f0/python/pyarrow/tests/test_dataset.py#L3004